paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -1323,143 +1323,88 @@ def check_if_instance_is_done(
-                )
-            )
-            return False
-        if not git_sha.startswith(status.git_sha):
-            print(
-                "  {}.{} on {} doesn't have the right sha yet: {}".format(
-                    service, instance, cluster, status.git_sha,
-                )
-            )
-            return False
-        if long_running_status.deploy_status not in [
-            "Running",
-            "Deploying",
-            "Waiting",
-        ]:
-            print(
-                "  {}.{} on {} isn't running yet: {}".format(
-                    service, instance, cluster, long_running_status.deploy_status,
-                )
-            )
-            return False
 
-        # The bounce margin factor defines what proportion of instances we need to be "safe",
-        # so consider it scaled up "enough" if we have that proportion of instances ready.
-        required_instance_count = int(
-            math.ceil(
-                instance_config.get_bounce_margin_factor()
-                * long_running_status.expected_instance_count
-            )
-        )
-        if required_instance_count > long_running_status.running_instance_count:
-            print(
-                "  {}.{} on {} isn't scaled up yet, "
-                "has {} out of {} required instances (out of a total of {})".format(
-                    service,
-                    instance,
-                    cluster,
-                    long_running_status.running_instance_count,
-                    required_instance_count,
-                    long_running_status.expected_instance_count,
-                )
-            )
-            return False
+    # Case: instance is stopped
+    if status.expected_instance_count == 0 or status.desired_state == "stop":
+        log.debug(f"{inst_str} is marked as stopped. Marked as deployed.")
+        return True
 
-        # `is not None` is necessary here because I'm adding the new attribute to the API response at the same time as I'm adding this reference.
-        if long_running_status.active_shas is not None:
-            active_git_shas = {g for g, c in long_running_status.active_shas}
+    # Case: bounce in-progress
+    if status.app_count > 1:
+        print(f"  {inst_str} is still bouncing, {status.app_count} versions running")
+        return False
 
-            if git_sha not in active_git_shas:
-                print(f"  {service}.{instance} on {cluster} not yet running {git_sha}.")
-                return False
+    # Case: bounce not yet started
+    # at this pt, only one app/version is active, and a 'sha' is [git_sha, config_sha],
+    # so active_shas[0][0] is the active git sha (not config sha)
+    if not git_sha.startswith(status.active_shas[0][0]):
+        print(
+            f"  {inst_str} hasn't started bouncing to the desired version yet; "
+            f"it is still on: {status.active_shas[0][0]}"
```

Comment:
I'd show all of the active shas here: `f"it is still running {', '.join({g for g, c in active_shas})}"`